### PR TITLE
Khj 22

### PR DESCRIPTION
**Pull request by RomanRB**

### CODE REVIEW

File: src/main/java/com/cargohub/entities/UserEntity.java [hunk 1]
```diff
@@ -2,17 +2,7 @@
...
-import javax.persistence.Entity;
-import javax.persistence.FetchType;
-import javax.persistence.GeneratedValue;
-import javax.persistence.Id;
-import javax.persistence.JoinColumn;
-import javax.persistence.JoinTable;
-import javax.persistence.ManyToMany;
-import javax.persistence.OneToMany;
-import javax.persistence.Table;
+import javax.persistence.*;
```
Comment (EvgeniyKoren) [inline src/main/java/com/cargohub/entities/UserEntity.java:5]: There is no use importing full package.

File: src/main/java/com/cargohub/dto/UserDto.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import lombok.Data;
 
 import java.util.Collection;
+import java.util.LinkedList;
```
Comment (EvgeniyKoren) [inline src/main/java/com/cargohub/dto/UserDto.java:6]: Please, remove unused import

File: src/main/java/com/cargohub/entities/UserEntity.java [hunk 3]
```diff
@@ -45,6 +35,7 @@
     @Column(nullable = false)
     private String phoneNumber;
 
+
```
Comment (EvgeniyKoren) [inline src/main/java/com/cargohub/entities/UserEntity.java:38]: Please, remove redundant operator - empty line.

File: src/main/java/com/cargohub/repositories/BillingDetailsRepository.java [hunk 4]
```diff
@@ -4,4 +4,7 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface BillingDetailsRepository extends JpaRepository<BillingDetailsEntity, Long> {
+
+
+
```
Comment (EvgeniyKoren) [inline src/main/java/com/cargohub/repositories/BillingDetailsRepository.java:9]: Please, remove redundant operator - empty line.

File: src/main/java/com/cargohub/security/jwt/JwtTokenProvider.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    private long validityInMilliseconds;
+
+
+    @Autowired
+    private UserDetailsService userDetailsService;
+
+    @Bean
+    public BCryptPasswordEncoder passwordEncoder() {
+        BCryptPasswordEncoder bCryptPasswordEncoder = new BCryptPasswordEncoder();
+        return bCryptPasswordEncoder;
```
Comment (EvgeniyKoren) [inline src/main/java/com/cargohub/security/jwt/JwtTokenProvider.java:42]: Why it removed from Config class?

File: src/test/java/com/cargohub/services/impl/UserServiceImplTest.java [hunk 6]
```diff
@@ -89,7 +89,7 @@ void setUp() {
     }
 
     private List<BillingDetailsDto> getBillingDetailsDto() {
-        List<BillingDetailsDto> listBillingDetailsDto = new ArrayList<>();
+        ArrayList<BillingDetailsDto> listBillingDetailsDto = new ArrayList<>();
```
Comment (EvgeniyKoren) [inline src/test/java/com/cargohub/services/impl/UserServiceImplTest.java:92]: Should it be changed to ArrayList?

File: src/main/java/com/cargohub/security/jwt/JwtTokenProvider.java (hunk 5 above)
Comment (RomanRB) [reply src/main/java/com/cargohub/security/jwt/JwtTokenProvider.java:42]: This bean moved to JwtTokenProvider class during the bug fixing. If it's necessary this bean can be moved back to Config.

File: src/test/java/com/cargohub/services/impl/UserServiceImplTest.java (hunk 6 above)
Comment (RomanRB) [reply src/test/java/com/cargohub/services/impl/UserServiceImplTest.java:92]: Thanks for noticing

File: src/main/java/com/cargohub/controllers/AuthenticationController.java [hunk 7]
```diff
@@ -1,35 +1,80 @@
...
+    @PostMapping("/registration")
     public ResponseEntity register(@RequestBody RegistrationModel registrationModel) {
         UserDto userDto = modelMapper.map(registrationModel, UserDto.class);
         userService.createUser(userDto);
 
         return new ResponseEntity(HttpStatus.CREATED);
     }
 
+    @PostMapping("/login")
+    public ResponseEntity login(@RequestBody AuthRequestModel requestDto) {
```
Comment (Dimon-Mvp) [inline src/main/java/com/cargohub/controllers/AuthenticationController.java:53]: I think, that you need to transfer logic to the service

**Review by Dimon-Mvp — CHANGES_REQUESTED** [review]

**Review by Dimon-Mvp — COMMENTED** [review]
Please transfer logic to the service

**Review by Dimon-Mvp — COMMENTED** [review]
Please transfer logic to the service

File: src/main/java/com/cargohub/controllers/AuthenticationController.java (hunk 7 above)
Comment (EvgeniyKoren) [reply src/main/java/com/cargohub/controllers/AuthenticationController.java:53]: @Dimon-Mvp , why do you think that?

Comment (RomanRB) [reply src/main/java/com/cargohub/controllers/AuthenticationController.java:53]: @Dimon-Mvp, what kind of logic and what kind of service do you mean?

File: src/main/java/com/cargohub/UserServiceApp.java [hunk 8]
```diff
@@ -1,12 +1,7 @@
...
 
-import org.modelmapper.ModelMapper;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.boot.builder.SpringApplicationBuilder;
-import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;
-import org.springframework.context.annotation.Bean;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 
 @SpringBootApplication
```
Comment (Dimon-Mvp) [inline src/main/java/com/cargohub/UserServiceApp.java:6]: What is this class?

**Review by Dimon-Mvp — APPROVED** [review]

File: src/main/java/com/cargohub/controllers/AuthenticationController.java [hunk 9]
```diff
@@ -1,35 +1,80 @@
...
+        try {
+            String email = requestDto.getEmail();
+            authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(email, requestDto.getPassword()));
+            UserDto user = userService.getUser(email);
+
+            if (user == null) {
```
Comment (SoftMikle) [inline src/main/java/com/cargohub/controllers/AuthenticationController.java:59]: Following throw is unnecessary, it duplicates the same in user service and is unreachable

**Pull request merged**